docs/basic-component.md

```diff
@@ -0,0 +1,247 @@
+  const [hasSyrup, setHasSyrup] = useState<boolean>(false)
+
+  const handleCustomFunction = () => {
+    onCustomFunction()
+    setHasSyrup(!hasSyrup)
+    return true
+  }
+
+  return (
+    <div
+      className={classnames(styles.pancakeStack, classNameOverride, {
+        [styles.someClass]: isBooleanProp,
+      })}
+      onSomething={handleCustomFunction}
+      {...props}
+    >
+      {children}
+      {doSomething(hasOptionalBooleanProp) && <SubComponent />}
+    </div>
+  );
+}
+
+PancakeStack.displayName = "PancakeStack"
+```
+
+### Props
+
+```tsx
+import { HTMLAttributes } from "react"
+
+export interface PancakeStackProps extends Omit<HTMLAttributes<HTMLDivElement>, "className"> {
+  children: React.ReactNode
+  classNameOverride?: string
+  isBooleanProp: boolean
+  hasOptionalBooleanProp?: boolean
+  onCustomFunction: () => void
+}
+```
+
+- Create and export an `interface` for your props in the format of `{PascalComponentName}Props`
+  - You can restrict a prop's type signature by redeclaring it (eg. changing a prop from `string | undefined` to `string`)
+  - If you need to change the type signature of an extended prop, ensure you `Omit` it otherwise it will become `never` (eg. changing a prop from `string | undefined` to `number`)
+- `Omit` `"className"` as we re-add it with an alias (`classNameOverride`)
+- Extend the native attributes of the closest HTML element of your component (eg. `<section>` will use `<div>` attributes)
+
+```tsx
+// Extending <section>
+import { HTMLAttributes } from 'react'
+export type SectionProps = Omit<HTMLAttributes<HTMLDivElement>, "className">
+
+// Extending <button>
+import { ButtonHTMLAttributes } from 'react'
+export type ButtonProps = Omit<ButtonHTMLAttributes<HTMLButtonElement>, "className">
+```
+
+- If your component is mainly extending another component, extend the props of that component and spread the props
+  - This will prevent the need for updating the consuming component with any new props in the child
+  - You don't need to extend HTML attributes as they would come from the child props (this also ensures consistency)
+
+```tsx
+export interface NewComponentProps extends PancakeStackProps {}
+
+export const NewComponent: React.VFC<NewComponentProps> = ({ ...props }) => <PancakeStack {...props} />
+```
+
+- Declare the `children` prop if you require it
+  - Usually it is `React.ReactNode`, however you can customise it to your needs
+- Add `classNameOverride` to replace the omitted `className`
+- Prefix boolean props with `is` or `has`
+- Declare a default value for optional boolean props
+  - Unless you specifically have a need to differentiate between `false` and `undefined`, this allows you to have the type safety of only needing to cater for a boolean value in any usages (eg. in a util)
+  - Aim to name the prop so that the default value is `false` which will allow the consumer to only need to include `isBoolean` as opposed to needing to negate it using `isBoolean={false}`
```

Comment:
Not sure if this is needed if omitting the prop will effectively have false unless I've missed something. I'm thinking along the lines of how `<Component isBoolean />` is equivalent to `<Component isBoolean={true} />`

Reply:
Probably just needs rewording. The sentence is meant to mean that we want the consumers to only need to do `<Component />` (default false) and `<Component isBoolean />` (set to true) as opposed to `<Component />` (default true) `<Component isBoolean={false} />` (set to false)